engine-precompiles/src/native.rs

```diff
@@ -553,6 +591,10 @@ impl<I: IO> Precompile for ExitToEthereum<I> {
         let mut input = input;
         let flag = input[0];
         input = &input[1..];
+        #[cfg(not(feature = "ext-connector"))]
+        let eth_connector_account_id = self.current_account_id.clone();
+        #[cfg(feature = "ext-connector")]
+        let eth_connector_account_id = get_eth_connector_contract_account(&self.io)?;
```

Comment:
I wonder if it makes sense to try an simplify some of this conditional compilation by having the implementation of `get_eth_connector_contract_account` depend on the feature (if it present read from state, otherwise use current account id).

This will make the definition of `get_eth_connector_contract_account` more complicated, but simplify all the call-sites.

Reply:
I'm not sure that it makes sense. We should add two implementations for the `ExitToNear` and `ExitToEthereum` to get access to the `self.current_account_id` field. So we will have two identical methods instead of one function.